ONSdigital/dp-import-api
# Prevent app from shutting down

### What

Prevent the app from shutting down due to connection errors to external services (e.g. kafka) on start up and whilst app is running.

The service will later include a health checker which will enable the platform to send a os signal interrupt or terminate to shutdown the service if the platform deems it necessary.

Also includes refactoring the code to use dp-rchttp library instead of go-ns/rchttp and general update of dependencies - including goconvey. Moved http listen and serve logic out of main function to api package.

### How to review

- Check service has no references to go-ns/rchttp 
- Check service does not shutdown on start up if external services it depends on are unavailable (e.g. kafka)
- Check service logs out an error if connection to external service is lost but does NOT cause a graceful shutdown (only os signal interrupt and terminate should force a graceful shutdown)

### Who can review

Anyone
